packages/blade/src/components/Radio/__tests__/RadioGroup.web.test.tsx

```diff
@@ -0,0 +1,189 @@
+        </Radio>
+        <Radio helpText="Mango help" value="mango">
+          Mango
+        </Radio>
+      </RadioGroup>,
+    );
+
+    expect(queryByText(helpText)).toBeInTheDocument();
+    expect(queryByText('Apple help')).toBeInTheDocument();
+    expect(queryByText('Mango help')).toBeInTheDocument();
+  });
+
+  it('should render errorText when validationState is set to error', () => {
+    const labelText = 'Select fruits';
+    const helpText = 'Select one';
+    const errorText = 'Invalid selection';
+
+    const { getAllByRole, queryByText } = renderWithTheme(
+      <RadioGroup
+        helpText={helpText}
+        errorText={errorText}
+        label={labelText}
+        validationState="error"
+      >
+        <Radio value="apple">Apple</Radio>
+        <Radio value="mango">Mango</Radio>
+        <Radio value="orange">Orange</Radio>
+      </RadioGroup>,
+    );
+
+    expect(queryByText(helpText)).not.toBeInTheDocument();
+    expect(queryByText(errorText)).toBeInTheDocument();
+    getAllByRole('radio', { hidden: true }).forEach((radio) => {
+      expect(radio).toBeInvalid();
+    });
+  });
+
+  it('should work in uncontrolled mode', async () => {
+    const user = userEvents.setup({ autoModify: false });
+    const labelText = 'Select fruits';
+    const fn = jest.fn();
+    const { getByLabelText, getByRole } = renderWithTheme(
+      <RadioGroup
+        name="fruits"
+        label={labelText}
+        defaultValue="apple"
+        onChange={({ value }) => fn(value)}
+      >
+        <Radio value="apple">Apple</Radio>
+        <Radio value="mango">Mango</Radio>
+        <Radio value="orange">Orange</Radio>
+      </RadioGroup>,
+    );
+
+    expect(getByRole('radio', { hidden: true, checked: true })).toHaveAttribute('value', 'apple');
+
+    await user.tab();
+    expect(getByLabelText('Apple')).toHaveFocus();
+    await user.keyboard('[ArrowDown]');
+    expect(getByLabelText('Mango')).toHaveFocus();
+    expect(fn).toBeCalledWith('mango');
+    await user.keyboard('[ArrowDown]');
+    expect(getByLabelText('Orange')).toHaveFocus();
+    expect(fn).toBeCalledWith('orange');
+    await user.keyboard('[ArrowDown]');
+    expect(getByLabelText('Apple')).toHaveFocus();
+    expect(fn).toBeCalledWith('apple');
+  });
+
+  it('should work in controlled mode', async () => {
+    const user = userEvents.setup();
+    const labelText = 'Select fruits';
+    const fn = jest.fn();
+    const Example = () => {
+      const [value, setValue] = React.useState('apple');
+      return (
+        <>
+          <RadioGroup
+            label={labelText}
+            value={value}
+            onChange={({ value }) => {
+              setValue(value);
+              fn(value);
+            }}
+          >
+            <Radio value="apple">Apple</Radio>
+            <Radio value="mango">Mango</Radio>
+            <Radio value="orange">Orange</Radio>
+          </RadioGroup>
+          <p data-testid="values">{value}</p>
```

Comment:
What is this for?

Reply:
Just a dummy text node to assert to value against the controlled radio.